### ISSUE TITLE
About the Subset out anatomical regions in Seurat v3.1.2

### ISSUE THREAD

**Issue by BioAIEvolu** [satijalab/seurat#2454]
Hi~I meet some problem in Subset out anatomical regions

I use the dataset: Mouse Brain Section (Coronal), and run sapceranger successfully,and get clustering_visualization img.

![clustering_visualization.png](https://user-images.githubusercontent.com/31410378/71407829-18f69b80-2677-11ea-9f14-49231b1f18a2.png)

Then, everything run smoothly before Subset out anatomical regions.Howerver,I choose the the cluster of cortex for subsetting like this:
```
#Subset out anatomical regions
cortex <- subset(brain, idents = c(0, 1, 2, 5, 10, 11, 12 ))
# now remove additional cells, use SpatialDimPlots to visualize what to remove
# SpatialDimPlot(cortex,cells.highlight = WhichCells(cortex, expression = image_imagerow > 400 |
# image_imagecol < 150))
cortex <- subset(cortex, imgine_imagerow > 400 | image_imagecol < 150, invert = TRUE)
cortex <- subset(cortex, image_imagerow > 275 & image_imagecol > 370, invert = TRUE)
cortex <- subset(cortex, image_imagerow > 250 & image_imagecol > 440, invert = TRUE)
```
then, the error occur:
```
> #Subset out anatomical regions
> cortex <- subset(brain, idents = c(0, 1, 2, 5, 10, 11, 12 ))
> # now remove additional cells, use SpatialDimPlots to visualize what to remove
> # SpatialDimPlot(cortex,cells.highlight = WhichCells(cortex, expression = image_imagerow > 400 |
> # image_imagecol < 150))
> cortex <- subset(cortex, imgine_imagerow > 400 | image_imagecol < 150, invert = TRUE)
Error in FetchData(object = object, vars = expr.char[vars.use], cells = cells,  : 
  None of the requested variables were found: 
> cortex <- subset(cortex, image_imagerow > 275 & image_imagecol > 370, invert = TRUE)
Error in FetchData(object = object, vars = expr.char[vars.use], cells = cells,  : 
  None of the requested variables were found: 
> cortex <- subset(cortex, image_imagerow > 250 & image_imagecol > 440, invert = TRUE)
Error in FetchData(object = object, vars = expr.char[vars.use], cells = cells,  : 
  None of the requested variables were found: 
```

And I have try to use the solution: #1212 (comment) for subsetting on features that have a dash in their name. But it doesn't work.

So,I wonder  the structure of seurat may be the key to solve the problem. 
I tape:
```
cortex@images
```
and saw：
```
attr(,"class")
[1] "scalefactors"

Slot "coordinates":
                   tissue row col imagerow imagecol
AAACAATCTACTAGCA-1      1   3  43     1611     4170
AAACACCAATAACTGC-1      1  59  19     8315     2519
AAACAGAGCGACTCCT-1      1  14  94     2927     7679
```

I try change the imgine_imagerow into imagerow, and change the image_imagecol into the imagecol,but it doesn't work again.

And another question is : How can I  further segment based on exact positions? 

Because I don't know exactly about the coordinates of cortex.How the example in official website actually do in this step?

thanks!

**Comment by BioAIEvolu** [comment]
![clustering_visualization](https://user-images.githubusercontent.com/31410378/71408942-da62e000-267a-11ea-8b82-f5dde9bdaad0.png)

**Comment by LucaMarconato** [comment]
same problem here

**Comment by andrewwbutler** [comment]
Hi,

Please try replacing `image_imagerow` and `image_imagecol` with `anterior1_imagerow` and `anterior1_imagecol`. 

This is still under active development but the `do.identify` parameter of `SpatialDimPlot` should allow you to select cells manually or at least give you a sense of where you could start to draw some cutoffs.

**Comment by BioAIEvolu** [comment]
Thank you for your answer! Let me try again :-)

**Comment by FADHLyemen** [comment]
This solve it for me
```
SpatialDimPlot(cortex,cells.highlight = WhichCells(cortex, expression = slice1_imagerow > 10000
| slice1_imagecol > 100)) 
```
